src/Admin/AMPPlugins.php

```diff
@@ -0,0 +1,271 @@
+<?php
+/**
+ * Add new tab (AMP) in plugin install screen in WordPress admin.
+ *
+ * @package Ampproject\Ampwp
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+use AmpProject\AmpWP\Infrastructure\Conditional;
+use AmpProject\AmpWP\Infrastructure\Delayed;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use stdClass;
+use function get_current_screen;
+
+/**
+ * Add new tab (AMP) in plugin install screen in WordPress admin.
+ *
+ * @since 2.2
+ * @internal
+ */
+class AMPPlugins implements Conditional, Delayed, Service, Registerable {
+
+	/**
+	 * Assets handle.
+	 *
+	 * @var string
+	 */
+	const ASSET_HANDLE = 'amp-plugin-install';
+
+	/**
+	 * @var array List of AMP plugins.
+	 */
+	public static $plugins = [];
+
+	/**
+	 * Get the action to use for registering the service.
+	 *
+	 * @return string Registration action to use.
+	 */
+	public static function get_registration_action() {
+
+		return 'current_screen';
+	}
+
+	/**
+	 * Check whether the conditional object is currently needed.
+	 *
+	 * @return bool Whether the conditional object is needed.
+	 */
+	public static function is_needed() {
+
+		return is_admin();
+	}
+
+	/**
+	 * Fetch AMP plugin data.
+	 *
+	 * @return void
+	 */
+	public static function set_plugins() {
+
+		global $wp_filesystem;
+
+		AMPThemes::init_file_system();
+
+		$plugin_json = AMP__DIR__ . '/data/plugins.json';
+
+		if ( ! file_exists( $plugin_json ) ) {
+			return;
+		}
+
+		$json_data       = $wp_filesystem->get_contents( $plugin_json );
+		self::$plugins   = json_decode( $json_data, true );
+		$json_last_error = json_last_error();
+
+		if ( JSON_ERROR_NONE !== $json_last_error ) {
+			self::$plugins = [];
+		}
+	}
+
+	/**
+	 * Adds hooks.
+	 *
+	 * @return void
+	 */
+	public function register() {
+
+		$this->set_plugins();
+		$screen = get_current_screen();
+
+		if ( $screen instanceof \WP_Screen && in_array( $screen->id, [ 'plugins', 'plugin-install' ], true ) ) {
```

Comment:
Please add a `use WP_Screen` above.

```suggestion
		if ( $screen instanceof WP_Screen && in_array( $screen->id, [ 'plugins', 'plugin-install' ], true ) ) {
```